resourceGroup.tf

```diff
@@ -7,11 +8,11 @@ terraform {
         }
     }
 }
-
+#Provider block
 provider "azurerm" {
   features {}
 }
-
+#RG block
 resource "azurerm_resource_group" "my_terraform_demo" {
   location = "australiasoutheast"
   name = "my_terraform_demo"
```

Comment:
Turn the name and location into variables, so we can be flexible.
Create a variables.tf file that defines the variables, their types and description.
Then use that here.